snippets/list_to_str.md

```diff
@@ -0,0 +1,30 @@
+---
+title: stringify_list
+tags: utility,beginner
+firstSeen: 2021-08-03T023:27:00-04:00
+---
+
+The stringify_list function takes the contents of a list and
+returns a string of the list contents optionaly separated by a provided separator.
+
+- Sets `out_str` to an empty string
+- Loops over all ov the items in the provided list
+- Append the current list item to `out_str` + the provided seperator if any
+- If the current list item is the last item in the list, append it to `out_str` but do not append separator
+
+```py
+def stringify_list(lst, separator="") -> str:
+out_str = ""
+for i in range(len(lst)):
+  item = lst[i]
+  if i == len(lst) - 1:
+    out_str += str(item)
+  else:
+    out_str += str(item) + separator
+  return out_str
```

Comment:
this is pretty much a 1-liner in python. (For non str lists you can still map and convert them)

`seperator.join(lst)`

Reply:
Oh wow, I had no idea! Thanks for informing me anyways.